src/js/components/modals/AppModalComponent.jsx

```diff
@@ -419,6 +402,7 @@ var AppModalComponent = React.createClass({
               <CollapsiblePanelComponent title="Environment variables">
                 <OptionalEnvironmentComponent
                   errorIndices={state.errorIndices.env}
+                  generalError={this.getErrorMessage("env")}
```

Comment:
I'd like to pass `getErrorMessage` as a prop, just like we do for OptionalSettingsComponent (and later on the ContainerSettingsComponent).